### DISCUSSION TITLE
Language missing in repository languages

### DISCUSSION THREAD

**Discussion by meldafert** [community/community#110197] [Repositories] [answered]
### Select Topic Area

Question

### Body

I've noticed that in https://github.com/jusw85/mozlz4, it displays "100% Rust" in Languages.
This is somewhat misleading, as the wrapper is indeed 100% Rust (~200 lines) - however, the actual implementation is in C (~3000 lines), and can be found in the `mozlz4-sys` folder in the same repository, included using a [path dependency]{https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-path-dependencies}.
It appears that the displayed languages only take the top-level package into account, and completely ignore the path dependency. This does make sense for outside dependencies (eg. from crates.io), but not for internal ones.

**Comment by spenserblack** [comment, marked as answer]
The files in `mozlz4-sys/` are [marked as vendored](https://github.com/jusw85/mozlz4/blob/d003163712347b2aa5e88396281ee0cb511d4638/.gitattributes#L1). Vendored, generated, and documentation files are excluded from the language bar for the repository. See [how linguist works](https://github.com/github-linguist/linguist/blob/559a6426942abcae16b6d6b328147476432bf6cb/docs/how-linguist-works.md). The project's readme says

> [lz4.h](https://github.com/mozilla/gecko-dev/blob/0069a0cdc3cd136ea7411521219df6665f260be7/mfbt/lz4/lz4.h) and [lz4.c](https://github.com/mozilla/gecko-dev/blob/0069a0cdc3cd136ea7411521219df6665f260be7/mfbt/lz4/lz4.c) copied directly from Mozilla repo

Which is probably why the maintainer(s) decided to mark those files as vendored.

**Reply by meldafert** [reply]
Ah, thank you, that does make sense.
I had missed the comment about the Mozilla repo.
